db/migrations/000006_create_environment_reservations_table.up.sql

```diff
@@ -0,0 +1,14 @@
+CREATE TABLE IF NOT EXISTS environment_reservations(
+    id                serial PRIMARY KEY,
+    environment_id    integer REFERENCES environments (id),
+    name              text,
+    source            text,
+    url               text,
+    ttl               interval, 
+    mutex             boolean,
+    metadata          jsonb,
+    created_at        timestamp WITH TIME ZONE default now(),
+    updated_at        timestamp WITH TIME ZONE default now()
```

Comment:
I'm treating environment_reservations as a time-series/events table, so it's not technically holding state, it's just recording when someone has locked/released a environment so.
`SELECT * FROM Environment_table WHERE environment == environment.id ORDER_BY created_at DESC LIMIT 1`
should always give you the state (either it's a release, or lock has a TTL that is either expired or not).

In the case that we don't respect TTL at all or just don't have it, then if the most recent row is a `lock` then it's locked and it would have been released at whenever we see a release event.